src/tlo/methods/healthsystem.py

```diff
@@ -1924,18 +1989,33 @@ def _write_never_ran_hsi_event_counts_to_log_and_reset(self):
         self._never_ran_hsi_event_counts_cumulative += self._never_ran_hsi_event_counts_log_period
         self._never_ran_hsi_event_counts_log_period.clear()
 
+    def _write_hsi_event_no_blank_counts_to_log_and_reset(self):
```

Comment:
Need to add a doctring explaining what function `_write_hsi_event_no_blank_counts_to_log_and_reset()` does